protomap/proto.go

```diff
@@ -700,6 +726,137 @@ func protoFromPathsInternal(p proto.Message, vals map[*gpb.Path]any, valPrefix,
 	return nil
 }
 
+// createListField creates the entries of the repeated field fd within the protobuf message m, mapping the values within the val map.
+// valPrefix specifies the prefix to be applied to the paths within the vals map, protoPrefix specifies the prefix for the protobuf
+// message (if it is not the root), and fieldPath specifies the path to the field that is being mapped. ignoreExtras indicates whether
+// extra paths that do not exist in the message should be treated as errors.
+func createListField(m proto.Message, fd protoreflect.FieldDescriptor, fieldPath, valPrefix, protoPrefix *gpb.Path, vals map[*gpb.Path]any, ignoreExtras bool) (protoreflect.Value, error) {
+	keys := []map[string]string{}
+	keyPaths := []*gpb.Path{}
+
+	// We need to identify the keys that are within the list, as well as the data tree paths
+	// that they correspond to. We walk through the supplied values to determine which to process.
+	for p := range vals {
+		// Make the paths within the vals map absolute according to the supplied valPrefix.
+		absPath := &gpb.Path{
+			Elem: append(append([]*gpb.PathElem{}, valPrefix.Elem...), p.Elem...),
+		}
+		// Since the fieldPath is a schema path, then we need to compare just schema paths
+		// to avoid comparing the keys.
+		if !util.PathMatchesPathElemPrefix(schemaPath(absPath), schemaPath(fieldPath)) {
```

Comment:
Should this only populate fields when the ancestor keys of `fieldPath` also match with `absPath`? Is it just the last element's keys that need to be removed for this check?

Reply:
In the case that we're iterating down multiple lists, we know that `vals` can only contain the subtree that we are in, at the point that we get to `a/list-one[key=a]/b/list-two[key=b]` then the only possible key of `list-one` must be `a` because otherwise we're in a different subtree -- so it seemed safe to just strip all keys here.

Reply:
No problem!